### ISSUE TITLE
[New option] shorthand first for sort-keys

### ISSUE THREAD

**Issue by anikethsaha** [eslint/eslint#13269]
**What rule do you want to change?**
`sort-keys`

**Does this change cause the rule to produce more or fewer warnings?**
More

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option. default false

**Please provide some example code that this change will affect:**
New options

```js
shortHand: {
   type: boolean,
   default: false
}
```

```js
const obj = {
  foo,
  a: 1,
  b: 2,
  bar // <-- error here, `bar` should come before non-shorthand, i.e with `foo`
};
```

```js
const obj = {
  foo,
  bar, // <-- error here, `bar` should come before `foo`
  a: 1,
  b: 2
};
```

Mainly when `shortHand` options will be `true`, all keys will be divided into two groups.

- `non-shorthand`
- `shorthand`

The ordering (`asc`, `desc`) will work in each groups (shorthand and non-shorthand groups) .
so with above example you can see, that there is no error stating, `'a' should come before 'bar'` cause groups will be independent of other group.

`motivation` : [airbnb styleguide](https://github.com/airbnb/javascript#objects--grouped-shorthand)

**What does the rule currently do for this code?**
doesnt report for shorthand below non-shorthand.

**What will the rule do after it's changed?**
it will report if shorthand are used after non-shorthand

**Are you willing to submit a pull request to implement this change?**
yes,
